### ISSUE TITLE
$mainAdminId should be configurable without needing to edit code

### ISSUE THREAD

**Issue by Zyten** [Zyten/moodle-tool_scormtrackeditor#2]

**Comment by danmarsden** [comment]
simplest option is to use something like the "is_siteadmin()" api on it's own, but the better solution would be to create your own capability within the plugin - given to admins by default and then you check that capability. That way the users of this plugin can decide to give the ability to update the tracks to certain users via the capabilty - eg not just the site admins who are usually a small set of people.

note - the hard-coded user id in there is a blocker for plugins db approval.

**Comment by Zyten** [comment]
Hi Dan, thank you so much for the feedback! 

Regarding this one, I aim to eventually make the plugin more modular, such that each teacher can only reset the SCORMs for their own courses with a GUI that's a bit more user-friendly. But this being my first Moodle plugin, I found the task daunting and decided to start with a simple form that's aimed at an Admin instead..

The hardcoded $mainAdminId was a poor attempt to make sure only one person (typically me) has access to this plugin to minimise possibility of abuse.

But assuming I go with creating my own capability, would it be possible to only assign this for the Main Administrator by default?

If that's not possible, using is_siteadmin() would still be acceptable for my use case I think. I just wanted to make it as restrictive as possible while the plugin is at it's current state.

**Comment by danmarsden** [comment]
yes - using capabilities is the preferred way to go - you can read more about that here:
https://moodledev.io/docs/apis/subsystems/access
